src/main/java/mods/railcraft/common/blocks/wayobjects/RoutingLogic.java

```diff
@@ -220,8 +238,8 @@ public AND(Condition a, Condition b) {
         }
 
         @Override
-        public boolean matches(IRoutingTile tile, EntityMinecart cart) {
-            return a.matches(tile, cart) && b.matches(tile, cart);
+        public int evaluate(IRoutingTile tile, EntityMinecart cart) {
+            return (a.evaluate(tile, cart) != 0 ? b.evaluate(tile, cart) : valueForBool(false));
```

Comment:
> The only issue is that it doesn't mesh well with redstone wire signal drop, though you could always set the low bit(s) to 1s to give a little bit of distance to work with.

That is not an issue. vanilla redstone comparator sometimes outputs 1 and works fine.